plugin/transfer/README.md

```diff
@@ -0,0 +1,80 @@
+# transfer
+
+## Name
+
+*transfer* - answer zone transfers requests for compatible authoritative
+plugins.
+
+## Description
+
+This plugin answers zone transfers for authoritative plugins that implement
+`transfer.Transferer`.
+
+Transfer answers AXFR requests and IXFR requests with AXFR fallback if the
+zone has changed.
+
+Notifies are not supported.
```

Comment:
meaning... a plugin still needs to send those by itself? How does it know where to send them ?

Reply:
It's not implemented yet.  I was hoping to implement notify in a separate PR, to keep the PR short.  Is that OK, or do you want notifies to be handled in this PR too?

Reply:
What are you thinking here? How will `transfer` know the zone has changed? Periodic polling?

We could bring back the `watch` infrastructure, which allowed plugins to push an event down a channel ... though it would likely need some refactoring to work in this internal manner (`transfer` subscribing to updates of zones from other plugins, rather than clients subscribing to updates of names).

Reply:
I was thinking polling because it seems to me like a simpler/cleaner solution.

In a polling design, a plugin would implement a `Serial(zone) serial` interface, which would be trivial to implement - just returning the current serial for the zone. Transfer would periodically poll each plugin in Tranferers for each zone, then send the notifies to listed hosts if serials change (as compared to a state maintained by the transfer plugin).

In a channel push design, a plugin would need to add code to their existing areas that update their serial.  This would not be as trivial to implement from the plugin perspective (code spread out a bit more).  There may be a cleaner way to do a push design that I'm not seeing right now.

Channel push would get notifies send more timely, but how timely are notifies expected to be? It could also be more spammy, if thats a concern.

Reply:
Yeah, that's a reasonable argument.